### DISCUSSION TITLE
Persistent custom dialog?

### DISCUSSION THREAD

**Discussion by bvbellomo** [quasarframework/quasar#12130] [General - Components / Directives / etc] [answered]
How do we use a custom component
https://quasar.dev/quasar-plugins/dialog#invoking-custom-component
with behavior similar to the persistent: true in the basic examples?

Everything I've tried, 'ESC' or clicking off the dialog just gets rid of it.

**Comment by pdanpdan** [comment]
pass `persistent: true` - all dialog props can be passed to `dialog.create` in the object

**Reply by bvbellomo** [reply]
Here is my setup:

  setup(){
    const $q = useQuasar();
    function showTimeoutDialog(){
      $q.dialog({
        component: SessionTimeout,
        persistent: true
      }).onOk(() => {
        // console.log('OK')
      }).onCancel(() => {
        // console.log('Cancel')
      }).onDismiss(() => {
        // console.log('I am triggered on both OK and Cancel')
      })}
    return {
      showTimeoutDialog
    }
  }

The "persistent: true" does not change anything, although it works great with the basic dialogs.

Oddly enough, the 'Esc' button does not close the dialog, but clicking off of it does.

**Reply by pdanpdan** [reply]
Sorry, my bad - with custom components put the props on the QDialog in custom component (or v-bind attrs in custom component and pass it in .create)

**Comment by bvbellomo** [comment]
I don't understand.  

My custom component's template (copied straight from the example) has:
<q-dialog ref="dialogRef" @hide="onDialogHide" persistent:true>

Is this what you mean?  VSCode (Volar) picks up 'persistent' with intellisense, but again, this does not appear to do anything.

**Comment by pdanpdan** [comment]
`<q-dialog ref="dialogRef" @hide="onDialogHide" persistent>`

Please make a codepen

**Reply by bvbellomo** [reply]
I will set one up, however my code is identical to the example

**Comment by bvbellomo** [comment, marked as answer]
This turned out to be "persistent:true" vs "persistent"

**Reply by hawkeye64** [reply]
If a property is on the component, it is assumed to be true (when Boolean). `persistent` is the same as `:persistent="true"`
